File: server/PoultryFarmManager.Core/Models/Finance/Vendor.cs
using System;

namespace PoultryFarmManager.Core.Models.Finance;

public class Vendor : DbEntity
{
    public string Name { get; set; } = string.Empty;
    public string? Location { get; set; }

    public Guid ContactPersonId { get; set; }
    public Person? ContactPerson { get; set; }
}

File: client/webapp/src/types/vendor.ts
import type { Person } from "./person";

export interface Vendor {
    id: string;
    name: string;
    location?: string;
    contactPersonId: string;
    contactPerson?: Person;
}

export interface NewVendor {
    name: string;
    location?: string;
    contactPersonId: string;
}

export interface UpdateVendor {
    name?: string;
    location?: string;
    contactPersonId?: string;
}
